mail/permissions.py

```diff
@@ -18,3 +21,34 @@ def has_permission(self, request, view):
         Returns True if the user has the 'can_message_learners' permission.
         """
         return has_permission(request.user, Permissions.CAN_MESSAGE_LEARNERS)
+
+
+class UserCanMessageCourseTeamPermission(BasePermission):
+    """
+    Permission class indicating permission to send a message to a course team.
+    """
+
+    def has_object_permission(self, request, view, obj):
+        """
+        Returns True if the user has permission to send a message to a course team.
+        Args:
+            request (Request): DRF request object
+            view (View): DRF view object
+            obj (Course): Course object
+        Returns:
+            boolean
+        """
+        user = request.user
+        program = obj.program
+        # Make sure the user has an enrollment in the course's program
+        if not ProgramEnrollment.objects.filter(user=user, program=program).exists():
+            return False
+        # Make sure the user has paid for any course run for the given course
+        edx_user_data = CachedEdxUserData(user)
+        mmtrack = MMTrack(
+            user,
+            program,
+            edx_user_data
+        )
+        course_run_keys = obj.courserun_set.values_list('edx_course_key', flat=True)
+        return any([mmtrack.has_paid(course_run_key) for course_run_key in course_run_keys])
```

Comment:
@giocalitri obviously this is doing a bunch of work that isn't needed, but this is the quickest way to check that a user has paid for a course implementation-wise. i'll open different issue to look into cases like these where we want to use specific functionality that exists in `MMTrack`, but we could avoid some of its overhead